doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+
+Doing this for each possible set $S$ involves $O(TL2^M)$ time complexity. We can significantly improve the algorithm to polynomial-time by computing the path of all sets simultaneously. The intuition here is to imagine standing at the first node and counting the number of subsets that will go one way, the number that will go the other, and the number that will go both (in the case of missing features). Because we assign different sized subsets different weights, we also need to distinguish the above numbers passing into each tree branch by their size. Finally, we also need to keep track of the proportion of sets of each size in each branch that contains a feature $i$ and the proportion that don't. Once all these sets have flowed down to the leaves of the tree, then we can compute the Shapley values. Doing this gives us $O(TLD^2)$ time complexity.
+
+**Pros**
+- Very fast for a valuable category of models
+- Doesn't require access to the training data
+- The Shapley values are fairly distributed among the feature values
+- Shapley values can be easily interpreted and visualized
+
+**Cons**
+- Only applies to Tree-based models
+- Uses an approximation of the interventional conditional expectation instead of computing it directly
+
+#### Interventional Tree SHAP
+
+The interventional TreeSHAP method takes a different approach. Suppose we sample a reference data point, $r$, from the training dataset. Let $F$ be the set of all features. For each feature, $i$, we then enumerate over all subsets of $S\subset F \setminus \{i\}$. If a subset is missing a feature, we replace it with the corresponding one in the reference sample. We can then compute $f(S)$ directly for each coalition $S$ to get the Shapley values. One major difference here is combining each $S$ and $r$ to generate a data point. Enforcing independence of the $S$ and $F\setminus S$ in this way is known as intervening in the underlying data distribution and is where the algorithm's name comes from. Note that this breaks any independence between features in the dataset, which means the data points we're sampling won't be realistic.
+
+For a single Tree and sample $r$ if we iterate over all the subsets of $S \subset F \setminus \{i\}$, the interventional TreeSHAP method runs with $O(M2^M)$. Note that there are two paths through the tree of particular interest. The first is the instance path for $x$, and the second is the sampled/reference path for $r$. Computing the Shapley value estimate for the sampled $r$ will involve replacing $x$ with values of $r$ and generating a set of perturbed paths. Instead of iterating over the sets, we sum over the paths. Doing so is faster as many of the routes within the tree have overlapping components. We can compute them all at the same time instead of one by one. Doing this means the Interventional TreeSHAP algorithm obtains $O(LD)$ time complexity.
```

Comment:
Tree --> tree

Reply:
Same comment as above, probably no need to go into detail of how the complexity is decreased from naive to polynomial.